core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -571,71 +574,117 @@ public static Connection getDBConnection(DataSource dataSource) throws SQLExcept
     }
 
     public static void closeConnection(Connection dbConnection) {
-
-        if (dbConnection != null) {
-            try {
-                dbConnection.close();
-            } catch (SQLException e) {
-                log.error("Database error. Could not close statement. Continuing with others. - " + e.getMessage(), e);
-            }
-        }
+        close(dbConnection);
     }
 
-    private static void closeResultSet(ResultSet rs) {
+    // Although using generics is a bit more cryptic to the uninitiated, this method ensures type safety
```

Comment:
This is opinionated :) Please make sure you only add comments if it helps other devs to better understand the context.